src/daemon/rpmostreed-transaction-types.c

```diff
@@ -191,13 +177,15 @@ package_diff_transaction_execute (RpmostreedTransaction *transaction,
 
   sysroot = rpmostreed_transaction_get_sysroot (transaction);
   upgrader = rpmostree_sysroot_upgrader_new (sysroot,
-					     self->osname,
-					     upgrader_flags,
-					     cancellable,
-					     error);
+                                             self->osname,
+                                             upgrader_flags,
+                                             cancellable,
+                                             error);
   if (upgrader == NULL)
     goto out;
 
+  origin = rpmostree_sysroot_upgrader_get_origin (upgrader);
```

Comment:
I'd feel better if this was `rpmostree_sysroot_upgrader_`*dup*`_origin()`, then when we've changed the new origin, we call `rpmostree_sysroot_upgrader_set_origin()` or so.  In practice right now it may not matter because we always parse a new origin, but I could imagine caching that origin rather than re-parsing it for every txn.

Or alternatively, we define that the upgrader creates a new origin, and we simply drop `rpmostree_sysroot_upgrader_set_origin()`?  (Does anything call it in your new code?)